### ISSUE TITLE
no way to fail out of an Promise's `always` block

### ISSUE THREAD

**Issue by catmando** [opal/opal#1967]
```ruby
p = Promise.new
p.always { |x| Promise.error('broken') }.fail { |z| puts "#{z}" }
```

does not work, you can do `raise 'broken'` but what if you want to pass a non exception object on to the failure block?

**Comment by hmdne** [comment]
This should be fixed with #2220

**Comment by hmdne** [comment]
It works in both V1 and V2. Either it didn't work before, which is possible, or you forgot to call `p.resolve` or `p.reject`.
